tools/build_config/Jenkinsfile

```diff
@@ -1,118 +1,182 @@
 #!groovy
 
 def post_github_status(String state, String message) {
-    if (env.PR_STATUSES_URL) {
-        withCredentials([string(credentialsId: 'GitHub_API_Token',
-                variable: 'api_token')]) {
-            sh """
-                curl -H "Authorization: token ${api_token}" \
-                    --request POST \
-                    --data '{"state": "${state}", \
-                            "description": "${message}", \
-                            "target_url": "$BUILD_URL", \
-                            "context": "$JOB_BASE_NAME"}' \
-                    $PR_STATUSES_URL > /dev/null
+  // Non-PR builds will not set PR_STATUSES_URL - in which case we do not
+  // want to post any statuses to Git
+  if (env.PR_STATUSES_URL) {
+    script {
+      withCredentials([string(credentialsId: 'GitHub_API_Token',
+          variable: 'api_token')]) {
+        if (isUnix()) {
+          sh """
+            curl -H "Authorization: token ${api_token}" \
+              --request POST \
+              --data '{"state": "${state}", \
+                "description": "${message}", \
+                "target_url": "$BUILD_URL", \
+                "context": "$JOB_BASE_NAME"}' \
+              $PR_STATUSES_URL > /dev/null
             """
         }
+        else {
+          powershell """
```

Comment:
should this be in a `isWindows` check?

Reply:
As far as I know, there's no such thing https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#isunix-checks-if-running-on-a-unix-like-node. Jenkins only has a `isUnix` or `not isUnix`

Reply:
That's frustrating... we'll need finder-grained resolution than that with iDAaaS, SCARF and Mac builds.